src/xiaomi_ble/parser.py

```diff
@@ -525,7 +534,9 @@ def obj1007(
         if device_type in ["MJYD02YL", "MCCGQ02HL"]:
             # 100 means light, else dark (0 or 1)
             # MCCGQ02HL might use obj1018 for light sensor, just added here to be sure.
-            return {"light": 1 if illum == 100 else 0}
+            device.update_predefined_binary_sensor(
+                BinarySensorDeviceClass.LIGHT, True if illum == 100 else False
```

Comment:
This could be wrapped in bool instead

Reply:
Or indeed `illum == 100` would return a bool anyway.